hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/autoassign/AutoAssignChecker.java

```diff
@@ -146,7 +152,8 @@ private int runTransactionalAssignment(final TargetFilterQuery targetFilterQuery
                             targetFilterQuery.getAutoAssignWeight().orElse(null), PAGE_SIZE);
                     final int count = deploymentRequests.size();
                     if (count > 0) {
-                        deploymentManagement.assignDistributionSets(deploymentRequests, actionMessage);
+                        systemSecurityContext.runAsSystemAsTenantAsUser(() -> deploymentManagement.assignDistributionSets(deploymentRequests, actionMessage),
```

Comment:
Not sure if this is intended: With this change you won't be able to distinguish between a manual assignment and an automatic assignment from the auto scheduler. It will always be the poor guy who created the filter

Reply:
that is true, do you think it makes sense to prepend something like `autoassign:` or something like that to the username?

Reply:
would it make sense to track the enabling of the autoassignment, something like an extra field `auto_assign_created_by` and use that user instead of the creator of the filter?

Reply:
what do you think about creating a field `auto_assign_triggered_by` or `auto_assign_enabled_by` in `TargetFilterQuery` table, and `action_triggered_by` filed in `Action` table?